tools/autograd/derivatives.yaml

```diff
@@ -403,12 +403,12 @@
 
 - name: index_add_(Tensor(a!) self, int dim, Tensor index, Tensor source) -> Tensor(a!)
   self: grad
-  source: grad.index_select(dim, index)
+  source: grad.index_select(dim, index).expand_as(source)
```

Comment:
When is this actually necessary? Only in the 0-dim case?

Reply:
yes that's right.  The shape is always correct otherwise, but since we treat both `()` and `(1,)` the same and allow allow expansion in a single direction, we need this.

Reply:
Ok sounds good.